src/status_im/contexts/network/events.cljs

```diff
@@ -0,0 +1,68 @@
+(ns status-im.contexts.network.events
+  (:require
+    [native-module.core :as native-module]
+    [re-frame.core :as re-frame]
+    [react-native.net-info :as net-info]
+    [status-im.common.data-confirmation-sheet.view :as data-confirmation-sheet]
+    [taoensso.timbre :as log]
+    [utils.re-frame :as rf]))
+
+(rf/reg-event-fx :network/set-syncing-on-mobile-network
+ (fn [{:keys [db]} [syncing-on-mobile-network?]]
+   {:db (update db
+                :profile/profile
+                assoc
+                :syncing-on-mobile-network?
+                syncing-on-mobile-network?
+                :remember-syncing-choice?
+                true)
+    :fx [[:json-rpc/call
+          [{:method     "wakuext_setSyncingOnMobileNetwork"
+            :params     [{:enabled syncing-on-mobile-network?}]
+            :on-success #(log/debug "successfully set syncing-on-mobile-network"
+                                    syncing-on-mobile-network?)
+            :on-error   #(log/warn "could not set syncing-on-mobile-network" %)}]]]}))
+
+(rf/reg-event-fx :network/check-expensive-connection
+ (fn [{:keys [db]}]
+   (when
+     (and (:network/is-expensive? db)
+          (:profile/profile db)
+          (not (get-in db [:profile/profile :remember-syncing-choice?])))
+     {:fx [[:dispatch
+            [:show-bottom-sheet
+             {:content (fn [] [data-confirmation-sheet/view])
+              :shell?  true
+              :theme   :dark}]]]})))
+
+(rf/reg-event-fx :network/network-info-changed
+ (fn [{:keys [db]} [{:keys [isConnected type details]}]]
+   (let [old-network-status  (:network-status db)
+         old-network-type    (:network/type db)
+         is-expensive?       (:isConnectionExpensive details)
+         connectivity-status (if isConnected :online :offline)]
+     (log/debug "[net-info]"
+                "old-network-status"  old-network-status
+                "old-network-type"    old-network-type
+                "connectivity-status" connectivity-status
+                "type"                type
+                "details"             details)
+     {:db (assoc db
+                 :network-status        (if isConnected :online :offline)
+                 :network/is-expensive? is-expensive?
+                 :network/type          type)
+      :fx [[:dispatch [:network/check-expensive-connection]]
+
+           [:network/notify-status-go [type is-expensive?]]]})))
+
+(re-frame/reg-fx
+ :network/listen-to-network-info
+ (fn []
+   (net-info/add-net-info-listener
+    #(re-frame/dispatch [:network/network-info-changed
+                         (js->clj % :keywordize-keys true)]))))
```

Comment:
A bonus would be not converting to CLJ because we don't store the response in the app-db:

- `isConnected`, only used in `(if isConnected :online :offline)`
- `type`, a primitive value, probably a number coming from status-go
- `details` is used to get `:isConnectionExpensive`, but this could be done using `oops/oget`.